src/zarr/storage/_fsspec.py

```diff
@@ -30,6 +32,45 @@
 )
 
 
+def _make_async(fs: AbstractFileSystem) -> AsyncFileSystem:
+    """Convert a sync FSSpec filesystem to an async FFSpec filesystem
+
+    If the filesystem class supports async operations, a new async instance is created
+    from the existing instance.
+
+    If the filesystem class does not support async operations, the existing instance
+    is wrapped with AsyncFileSystemWrapper.
+    """
+    import fsspec
+    from packaging.version import parse as parse_version
```

Comment:
why do we import these here?

Reply:
fsspec is imported here to follow the conventional of importing optional dependencies in the functions in which they are used, based on conversations from the obstore integration PR.

parse can be imported in the module scope so I'll change that.

Reply:
I moved the packaging.version.parse import in [`3b06c53` (#2774)](https://github.com/zarr-developers/zarr-python/pull/2774/commits/3b06c536cc5398a83b81eea7fc95abcdf3eaa37f)